src/baseApi.ts

```diff
@@ -55,6 +56,7 @@ export default abstract class BaseApi<ResourceName extends ResourceNames> {
         Authorization: `Bearer ${this.config.settings.accessToken ?? ''}`,
         'Content-type': 'application/json',
         ...(options?.headers as Record<string, string>),
+        ...this.config.settings.customRequestHeaders,
```

Comment:
Why you dont use `getCustomRequestHeaders()`?

Reply:
because getCustomRequestHeaders is Api class method and it is not available inside of any class extending BaseApi abstract class. otherwise we need to change constructor of Api class. also I can not move getter and setter to BaseApi class because `setCustomRequestHeaders` should be exposed outside and be accessible inside of app using CL